neuralmonkey/decoders/sequence_decoder.py

```diff
@@ -0,0 +1,258 @@
+"""Abstract class for decoding sequences left-to-right.
+
+Either for the recurrent decoder, or for the transformer decoder.
+
+The sequence decoder uses the while loop to get the outputs. Descendants should
+only specify the initial state and the while loop body.
+"""
+import math
+from typing import (NamedTuple, Union, Callable, Tuple, cast, Iterable, Type,
+                    List, Optional, Any)
+
+import numpy as np
+import tensorflow as tf
+
+from neuralmonkey.dataset import Dataset
+from neuralmonkey.decorators import tensor
+from neuralmonkey.logging import log
+from neuralmonkey.model.model_part import ModelPart, FeedDict
+from neuralmonkey.nn.utils import dropout
+from neuralmonkey.vocabulary import Vocabulary, START_TOKEN
+
+
+def extend_namedtuple(name: str, parent: Type,
```

Comment:
používá se tahle funkce někde?

Reply:
Nevim, refactor decoderu/sequence_decoderu jsi delal ty, ne?

Reply:
A koukam, ze to pouziva decoder.py aby si rozsiril feedables a histories, takze podle me je to ok.

Reply:
už to vidim. fajn